File: app/src/main/java/com/example/projectappdoctruyen/Adapter/ChapterWatchedAdapter.java
package com.example.projectappdoctruyen.Adapter;

import android.content.Context;
import android.content.Intent;
import android.text.Html;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.projectappdoctruyen.Activity.ReadStoryActivity;
import com.example.projectappdoctruyen.Model.Chapter;
import com.example.projectappdoctruyen.R;

import java.util.ArrayList;

public class ChapterWatchedAdapter extends RecyclerView.Adapter<ChapterWatchedAdapter.ViewHolder>  {
    Context context;
    ArrayList<Chapter> chapterArrayList_Wathched;

    public ChapterWatchedAdapter(Context context, ArrayList<Chapter> chapterArrayList_Wathched) {
        this.context = context;
        this.chapterArrayList_Wathched = chapterArrayList_Wathched;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_chapter_watched,parent,false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        Chapter chapter = chapterArrayList_Wathched.get(position);
        holder.txtChapterWatched.setText((chapter.getTitle()));

    }
    @Override
    public int getItemCount() {
        return chapterArrayList_Wathched.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder{
        TextView txtChapterWatched;
        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            txtChapterWatched = itemView.findViewById(R.id.txtChapterWatched);
            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    Intent intent = new Intent(context, ReadStoryActivity.class);
                    intent.putExtra("Chapter", chapterArrayList_Wathched.get(getPosition()));
                    context.startActivity(intent);
                }
            });
        }
    }
}

File: app/src/main/java/com/example/projectappdoctruyen/Adapter/ChapterWatchedAdapter.kt
package com.example.projectappdoctruyen.Adapter

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.projectappdoctruyen.Activity.ReadStoryActivity
import com.example.projectappdoctruyen.Model.Chapter
import com.example.projectappdoctruyen.R
import java.util.*

class ChapterWatchedAdapter(var context: Context, var chapterArrayList_Wathched: ArrayList<Chapter?>) : RecyclerView.Adapter<ChapterWatchedAdapter.ViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_chapter_watched, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val chapter = chapterArrayList_Wathched[position]
        holder.txtChapterWatched.text = chapter!!.title
    }

    override fun getItemCount(): Int {
        return chapterArrayList_Wathched.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var txtChapterWatched: TextView

        init {
            txtChapterWatched = itemView.findViewById(R.id.txtChapterWatched)
            itemView.setOnClickListener {
                val intent = Intent(context, ReadStoryActivity::class.java)
                intent.putExtra("Chapter", chapterArrayList_Wathched[position])
                context.startActivity(intent)
            }
        }
    }
}